cpp/stdlib.cc

```diff
@@ -120,6 +121,46 @@ void kill(int pid, int sig) {
   }
 }
 
+List<Str*>* listdir(Str* path) {
+  DIR* dirp = opendir(path->data());
+  if (dirp == NULL) {
+    throw Alloc<OSError>(errno);
+  }
+
+  auto* ret = NewList<Str*>();
+  while (true) {
+    errno = 0;
+    struct dirent* ep = readdir(dirp);
+    if (ep == NULL) {
+      if (errno == 0) {
+        break;
+      } else {
+        closedir(dirp);
```

Comment:
Nitpick: let's use the "early return" error style here, because we do above on line 126 (and throughout the codebase)